### DISCUSSION TITLE
Self-Hosted Rallly v3.0.0 (looking for beta testers)

### DISCUSSION THREAD

**Discussion by lukevella** [lukevella/rallly#729] [Announcements]
Hi everyone,

Before I make an official release I was hoping I would get some feedback from users who would be willing to test self-hosting Rallly v3.0.0.

## How to test it

You can run the new version by using the following docker image: 
```
lukevella/rallly:3.0.0-beta
``` 

If you're using the docker compose config from [rallly-selfhosted](https://github.com/lukevella/rallly-selfhosted), you can follow the [update instructions](https://github.com/lukevella/rallly-selfhosted#update-instructions).

## Config

The config remains mostly the same (see [rallly-selfhosted](https://github.com/lukevella/rallly-selfhosted)) except for the following changes:

* Removed `AUTH_REQUIRED` - The image is pre-built with this feature enabled
* Removed `DISABLE_LANDING`- This config is not required anymore since the landing page has been split into its own app

You don't have to remove these environment variables, they will just be ignored.

## What to expect

Self-hosted instances having billing settings disabled and all users will have access to Pro features on that self-hosted instance.

## Feedback

Please share feedback here if you successfully upgraded and let me know if you run into any issues.

**Comment by conor-f** [comment]
Upgraded successfully with no issues. Started with a fresh install of `latest` and followed the docs to set up behind Traefik with no issues.

However I won't be upgrading to it, since it forces authentication using SMTP with no option of just having simple authentication in the database, or removing authentication altogether that I could see. Is this the case?

As a potential change in the future, would any of the following be possible?
 - To bring back the option of handling auth yourself (e.g. using oauth or simple auth in Traefik)
 - Integration with [Apprise](https://github.com/caronc/apprise) for a more self-hosting friendly alternative to sending notifications instead of running an SMTP server
 - Allowing users to simply modify the database directly with a field like `is_verified`, `requires_auth=False` or a hashed `password` field
 - Disabling auth entirely
 
Additionally, I opened a PR in the rallly-selfhosted repo to expand on the Traefik set up as I noticed there were labels for `plausible` instead of `rallly` likely from a copy/paste!

Thanks for all the work you put into Rallly!

**Reply by lukevella** [reply]
Hey @conor-f, appreciate you taking the time to test it out. Thanks for that.

> However I won't be upgrading to it, since it forces authentication using SMTP with no option of just having simple authentication in the database, or removing authentication altogether that I could see. Is this the case?

Yes, the new version does require authentication to be enabled and this depends on having an SMTP server configured. I understand that there may be some users who may prefer to use Rallly without any email sending capabilities but it's simply not practical to support this use case as most of Rallly's current and future functionality relies on its ability to send out emails.

> To bring back the option of handling auth yourself (e.g. using oauth or simple auth in Traefik)

I don't plan on supporting this. It was only possible in the past because I hadn't implemented the ability to require authentication on the initial release of v2 and I've been waiting for the next major release to make it a default config for self-hosted instances.

> Integration with [Apprise](https://github.com/caronc/apprise) for a more self-hosting friendly alternative to sending notifications instead of running an SMTP server

Maybe, but unless a case is made for the value this will add to the project it's not where I'll be investing my own time so someone else would have to do this work if they want it.

> Allowing users to simply modify the database directly with a field like `is_verified`, `requires_auth=False` or a `hashed password field`

Catering specifically to users who have the technical ability to do this just isn't viable.

I understand these answers may not be what you want to hear. The project is in the process of transition from one that previously catered to hobbyists to one that focuses more on adding value. My priority at this point is to find ways to generate revenue because without this I won't be able to sustain working on this project as I have been.

Hope that makes sense and let me know if there's anything else you'd like me to clarify.

**Reply by conor-f** [reply]
Hey @lukevella 

Completely understandable and the answer I expected! Was just offering potential alternatives in case it was a straightforward solution.

Best of luck with Rallly!

**Comment by scoonie** [comment]
We are testing Self-Hosted Rallly v3.0.0 and some users get an error when finalising a poll
![WhatsApp Image 2023-07-26 at 14 07 37](https://github.com/lukevella/rallly/assets/48395518/71719cee-aa53-446a-b334-22bf620821e3)

I can finalise polls without any issue but some users cant, its quite strange

**Reply by lukevella** [reply]
Hi @scoonie, I'd need to see the error logged by the server to know what's wrong. If you don't see any errors in the server logs then you'd need to open the developer tools in your browser and check the network tab to see why the request is failing.

**Reply by scoonie** [reply]
Hi @lukevella there are no error logs in Apache. Is there somewhere else I should be looking for errors?

Developer options in Chrome gives the following error
![New Project](https://github.com/lukevella/rallly/assets/48395518/67528faf-4155-44fa-84a3-077d54e2e36d)

**Reply by lukevella** [reply]
Thanks @scoonie, I think I found a way to reproduce this so I’ll have a look at this tomorrow.

**Reply by lukevella** [reply]
There was a bug that was specific to polls that had all-day dates. I've pushed a fix for it now so if you pull the latest `3.0.0-beta` image it should be fixed now and you should be able to finalize dates that don't have times.

**Reply by scoonie** [reply]
Hi @lukevella this has resolved the issue, thank you very much for looking into this so quickly

**Comment by ep1cman** [comment]
Successfully went from 2.x.x to 3.0.0. The finalize feature worked as expected. Thanks for the great work!

**Comment by OysteinHoiby** [comment]
I've been testing out the beta today, and everything seems to work smoothly. I only have a question on language support; because it seems like the Norwegian locale is not activated?

**Reply by lukevella** [reply]
@OysteinHoiby the version available on docker hub right now was an earlier release. I'm working towards an official release of v3 on docker hub which will include the Norwegian locale. I'm hoping to release it this week.

**Reply by OysteinHoiby** [reply]
@lukevella Thank you for letting me know :) I really appreciate your work! Is there some specific things you would like us to test in v3 or are you looking for more general reports if we should stumble upon bugs/errors?
